hypershift-operator/controllers/nodepool/kubevirt/imagecacher.go

```diff
@@ -130,7 +154,7 @@ func (qi *cachedQCOWImage) CacheImage(ctx context.Context, cl client.Client, nod
 	// if no DV with the required hash was found
 	if len(dvName) == 0 {
 		logger.Info("couldn't find boot image cache DataVolume; creating it...")
-		dv, err := qi.createDVForCache(ctx, cl, nodePool, infraId)
```

Comment:
should the old cache be deleted before creating the new cache? Otherwise it seems like there's a potential for leaking cached images

For example, take this flow of events.

1. image is cached 
2. new release payload is set on nodepool
3. new image is cached
4. API call to cleanup old cached image fails due to api server temporarily not being available.

now it's possible that old DV will never get cleaned up.